tutor4points/core/views.py

```diff
@@ -94,16 +94,32 @@ def users(request, id):
 
     # get user that is specified by URL
     user = get_user_model().objects.get(pk=id)
-    if request.method == 'POST':
+    if request.method == 'POST' and 'email' in request.POST:
         form_update_profile = UpdateProfileForm(request.POST, request.FILES, instance=user)
         if form_update_profile.is_valid():
             form_update_profile.save()
     else:
         form_update_profile = UpdateProfileForm(instance=user)
 
+    if request.method == 'POST' and 'rating' in request.POST:#code for rating, move once 'paid and done' functionality is added.
+        form_rating = RateTutorForm(request.POST)
+        if form_rating.is_valid():
+            rating = form_rating.save(commit=False)
+            rating.given_to = user
```

Comment:
Later, when you move this part to the requests page, you would probably have to change this to the tutor from the request based on the request id or retrieving the tutor from the front end, but this works so far in terms of functionality.